draft-schwartz-ds-glue.md

```diff
@@ -79,62 +84,65 @@ $ORIGIN example.com.
 
-> QUESTION: Should we skip the virtual DNSKEY record, and construct the fake DS directly?  This would save 4-6 bytes per RR, but would lose the ability to reuse DNSKEY->DS construction codepaths (unchanged except for a new digest type).
+> QUESTION: Should we skip the virtual DNSKEY record, and construct the fake DS directly?  This would save 4-6 bytes per RRSet, but would lose the ability to reuse DNSKEY->DS construction codepaths (unchanged except for a new digest type).
 
 ## Interpretation
 
 Upon receiving a delegation response, resolvers implementing this specification SHALL compute the Adjusted Delegation Response as follows:
 
 1. Copy the delegation response.
-2. Reverse the encoding process of any DSGLUE records to reconstruct the source RRSets, all carrying the TTL of the DS RRSet.
+2. Reverse the encoding process of any DSGLUE records to reconstruct the source RRSets.
 3. Add each of these reconstructed RRSets to the Adjusted Delegation Response, replacing any RRSet with the same owner name and type.
 
+Note that a Source RRSet MAY be empty, indicating that there are no records of the corresponding type at this name.  After reconstructing an empty Source RRSet, recipients MUST remove any matching RRSets from the Adjusted Delegation Response and any glue cache, and MAY cache the negative result for the indicated TTL.
+
 Resolution then proceeds as usual, using the Adjusted Delegation Response.  When processing the DS RRSet, the recipient will verify the DS RRSIGs as usual, and abort the resolution as Bogus if DNSSEC validation fails.
 
 Resolvers that do not implement this specification will ignore the DSGLUE records due to the unrecognized algorithm.  Thus, these records are safe to use for both signed and unsigned child zones.
 
 Source Records reconstructed from DSGLUE SHOULD be processed exactly like ordinary unauthenticated glue records.  For example, they MAY be cached for use in future delegations but MUST NOT be returned in any responses (c.f. {{Section 5.4.1 of ?RFC2181}}).
 
-## Special case: RR Type = NSEC or NSEC3
-
-Normally, the absence of a particular record in a delegation response is not informative to a resolver.  The corresponding record might still exist in the child zone.  To inform the resolver that a particular RRSet is nonexistent for the purposes of delegation, the zone owner MAY place an NSEC or NSEC3 record in the delegation response.
+## Allowed RR types
 
-As with other glue records, an NSEC glue record only affects behavior during delegation following (see example in {{example-nsec}}).
+DSGLUE records are capable of containing any record type.  However, the meaning of certain record types (e.g. NSEC) is not yet clear in the DSGLUE context.  To avoid ambiguity, child zones MUST only publish DSGLUE records containing RR types that have been registered for use with DSGLUE ({{iana}}), and recipients MUST ignore DSGLUE records indicating unexpected record types.
```

Comment:
Explicitly list the RR types which should be supported (not always included): NS, A, AAAA.
Probably better to point to the new IANA registry described towards the end of the document.